docs/RFCs/0002-crashtracker-user-visable-log.md

```diff
@@ -0,0 +1,292 @@
+# RFC 0002: Crashtracker Structured Log Format (v1)
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in [IETF RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Summary
+This document describes version 1 of the crashinfo data format.
+
+## Motivation
+The `libdatadog` crashtracker detects program crashes.
+It automatically collects information relevant to the characterizing and debugging the crash, including stack-traces, the crash-type (e.g. SIGSIGV, SIGBUS, etc) crash, the library version, etc.
+This RFC establishes a standardized logging format for reporting this information.
+
+### Why structured json
+As a text-based format, json can be written to standard logging endpoints.
+It is (somewhat) human readable, so users can directly interpret the crash info off their log if necessary.
+As a structured format, it avoids the ambiguity of standard semi-structured stacktrace formats (as used by e.g. Java, .Net, etc).
+Due to the use of native extensions, it is possible for a single stack-trace to include frames from multiple languages (e.g. python may call C code, which calls Rust code, etc).
+Having a single structured format allows us to work across languages.
+
+## Proposed format
+A natural language description of the proposed json format is given here.
+An example is given in Appendix A, and the schema is given in Appendix B.
+
+### Required fields
+- `error`:
+    A structure of type `ErrorData`, described below.
+- `incomplete`:
+    Boolean `false` if the crashreport is complete (i.e. contains all intended data), `true` if there is expected missing data.
+    This can happen becasue the crashtracker is architected to stream data to an out of process receiver, allowing a partial crash report to be emitted even in the case where the crashtracker itself crashed during stack trace collection.
+    This MUST be set to `true` if any required field is missing.
+- `metadata`:
+    Metadata about the system in which the crash occurred:
+    - `library_name`
+    - `library_version`
+    - `family`
+    - `tags`:
+      A set of key:value pairs, representing any tags the crashtracking system wishes to associate with the crash.
+      Examples would include "hostname", "service", and any configuration information the system wishes to track.
+- `os_info`: 
+    The OS + processor architecture on which the crash occurred.
+- `stacktrace`: 
+    This represents the stack of the crashing thread.
+    See below for more details on how stacktraces are formatted.
+- `timestamp`:
+    The time at which the crash occurred, in ISO 8601 format.
+- `uuid`:
+    A UUID v4 which uniquely identifies the crash.
+- `version_id`:
+    An integer version id.
+
+### ErrorData
+- `additional_stacks`:
```

Comment:
Are these always threads? And are the `Identifier`s strings (of thread id's / names)?

If so, I think it would make sense to use the same format as RUM SDK's do where possible. An array `threads` of objects with attribute:
- `name` for the thread name (same as Identifier here? Can just be "Thread #<id>" which is what we do on iOS)
- `stack` for the stack trace (can be the structured format for the crash tracker)
- (optional, can leave out for the crash tracker format) `crashed` boolean for whether this thread crashed
- (optional, can leave out for the crash tracker format) `state` for thread state like RUNNING, BLOCKED etc. (different on each platform)

[example of an org2 iOS crash](https://app.datadoghq.com/error-tracking/unified?query=&issueId=616bbd8c-825d-11ee-9764-da7ad0900002&source=all&view=spans)

```
            "threads": {
              "type": "array",
              "description": "Description of each thread in the process when error happened.",
              "items": {
                "type": "object",
                "description": "Description of the thread in the process when error happened.",
                "required": ["name", "crashed", "stack"],
                "properties": {
                  "name": {
                    "type": "string",
                    "description": "Name of the thread (e.g. 'Thread 0').",
                    "readOnly": true
                  },
                  "crashed": {
                    "type": "boolean",
                    "description": "Tells if the thread crashed.",
                    "readOnly": true
                  },
                  "stack": {
                    "type": "string",
                    "description": "Unsymbolicated stack trace of the given thread.",
                    "readOnly": true
                  },
                  "state": {
                    "type": "string",
                    "description": "Platform-specific state of the thread when its state was captured (CPU registers dump for iOS, thread state enum for Android, etc.).",
                    "readOnly": true
                  }
                }
              }
            },
```

![image](https://github.com/user-attachments/assets/30d903e4-dc36-470b-9457-9a06972a2db7)

Reply:
Done